### PR TITLE
/vsicurl: add ANYSAFE & BEARER to auth methods

### PR DESCRIPTION
**Pull request by drons**

### CODE REVIEW

File: doc/source/user/configoptions.rst [hunk 1]
```diff
@@ -829,6 +829,19 @@ Networking options
...
 
+-  .. config:: GDAL_GSSAPI_DELEGATION
+      :since: 3.3
+      :choices: NONE, POLICY, ALWAYS
+
+      Set allowed GSS-API delegation. Relevant only with
+      :config:`GDAL_HTTP_AUTH=NEGOTIATE`.
+
+-  .. config:: GDAL_HTTP_BEARER
+      :since: 3.8
```
Comment (rouault) [inline doc/source/user/configoptions.rst:840]:
```suggestion
      :since: 3.9
```

File: doc/source/user/configoptions.rst [hunk 2]
```diff
@@ -829,6 +829,19 @@ Networking options
...
+
+      Set HTTP OAuth 2.0 Bearer Access Token to use for the connection. Must be used
```
Comment (rouault) [inline doc/source/user/configoptions.rst:842]: Is that enough ? I'm a bit confused by https://curl.se/libcurl/c/CURLOPT_HTTPAUTH.html which mentions "You can set the Bearer token to use with [CURLOPT_XOAUTH2_BEARER](https://curl.se/libcurl/c/CURLOPT_XOAUTH2_BEARER.html)."

Comment (drons) [reply doc/source/user/configoptions.rst:842]: I'll add a test to  [vsicurl.py](https://github.com/OSGeo/gdal/blob/master/autotest/gcore/vsicurl.py) a little later to remove doubts

Comment (drons) [reply doc/source/user/configoptions.rst:842]: To successful authorization we must provide both `GDAL_HTTP_AUTH=BEARER` and `GDAL_HTTP_BEARER`. If we comment out `unchecked_curl_easy_setopt(http_handle, CURLOPT_HTTPAUTH, CURLAUTH_BEARER);` than authorization will fail, even if we setup `CURLOPT_XOAUTH2_BEARER`.

**Comment by drons** [comment]
Thanks!

**Pull request merged**